From: SeongJae Park <sj@kernel.org>
Date: Fri, 27 Jun 2025 17:42:02 -0700
Subject: [PATCH] mm/damon: add write field to damon_access_report

This commit is for showing the first part of the idea for implementing
write-only access monitoring.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index ac108d6cbb85..0158bff9a672 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -112,6 +112,7 @@ struct damon_target {
  * @size:		The size of the reporting region.
  * @nr_accesses:	Number of detected accesses to the region.
  * @node_id:		NUMA node that made the access.
+ * @write:		Whether the access is write.
  *
  * @pid could be stale, and hence shouldn't be de-referenced.
  */
@@ -121,6 +122,7 @@ struct damon_access_report {
 	unsigned long size;
 	int nr_accesses;
 	int node_id;
+	bool write;
 /* private: */
 	unsigned long report_jiffies;	/* when this report is made */
 };
-- 
2.39.5

